website/blog/2025/01-03-scrape-crunchbase/index.md

```diff
@@ -0,0 +1,501 @@
+
+The structure of one of these files looks like this:
+
+![Sitemap second lvl](./img/sitemap_lvl_two.webp)
+
+The `lastmod` field is particularly important here. It allows tracking which companies have updated their information since the previous data collection. This is especially useful for regular data updates.
+
+### 1. Configuring the crawler for scraping
+
+To work with the site, we'll use [`CurlImpersonateHttpClient`](https://www.crawlee.dev/python/api/class/CurlImpersonateHttpClient), which impersonates a `Safari` browser. While this choice might seem unexpected for working with a sitemap, it's necessitated by Crunchbase's protection features.
+
+The reason is that Crunchbase uses [Cloudflare](https://www.cloudflare.com/) to protect against automated access. This is clearly visible when analyzing traffic on a company page:
+
+![Cloudflare Link](./img/cloudflare_link.webp)
+
+An interesting feature is that `challenges.cloudflare` is executed after loading the document with data. This means we receive the data first, and only then JavaScript checks if we're a bot. If our HTTP client's fingerprint is sufficiently similar to a real browser, we'll successfully receive the data.
+
+Cloudflare [also analyzes traffic at the sitemap level](https://developers.cloudflare.com/waf/custom-rules/use-cases/allow-traffic-from-verified-bots/). If our crawler doesn't look legitimate, access will be blocked. That's why we impersonate a real browser.
+
+To prevent blocks due to overly aggressive crawling, we'll configure [`ConcurrencySettings`](https://www.crawlee.dev/python/api/class/ConcurrencySettings).
+
+When scaling this approach, you'll likely need proxies. Detailed information about proxy setup can be found in the [documentation](https://www.crawlee.dev/python/docs/guides/proxy-management).
+
+We'll save our scraping results in `JSON` format. Here's how the basic crawler configuration looks:
+
+```python
+# main.py
+
+from crawlee.crawlers import ParselCrawler
+from crawlee.http_clients import CurlImpersonateHttpClient
+from crawlee import ConcurrencySettings, HttpHeaders
+
+async def main() -> None:
+    """The crawler entry point."""
+
+    concurrency_settings = ConcurrencySettings(max_concurrency=1, max_tasks_per_minute=50)
+
+    http_client = CurlImpersonateHttpClient(impersonate="safari17_0",
+                                            headers=HttpHeaders({
+                                                "accept-language": "en",
+                                                "accept-encoding": "gzip, deflate, br, zstd",
+                                                }))
+    crawler = ParselCrawler(
+        request_handler=router,
+        max_request_retries=1,
+        concurrency_settings=concurrency_settings,
+        http_client=http_client,
+        max_requests_per_crawl=30,
+    )
+
+    await crawler.run(['https://www.crunchbase.com/www-sitemaps/sitemap-index.xml'])
+
+    await crawler.export_data_json("crunchbase_data.json")
```

Comment:
Could you please use a consistent formatting? You can use configs from Crawlee for Python. This applies to all files.